pkgs/applications/editors/jetbrains/source/build.nix

```diff
@@ -22,45 +22,57 @@
   pkg-config,
   xorg,
 
-  version,
-  buildNumber,
   buildType,
-  ideaHash,
-  androidHash,
   jpsHash,
   restarterHash,
   mvnDeps,
 }:
 
 let
-  kotlin_2_0_21 = kotlin.overrideAttrs (oldAttrs: {
-    version = "2.0.21";
+  versions = builtins.fromJSON (builtins.readFile ./versions.json);
+  inherit (versions.${buildType}) version buildNumber;
+
+  ideaHash = versions.${buildType}.hash.intellij-community;
+  androidHash = versions.${buildType}.hash.android;
+
+  jbr = jetbrains.jdk-no-jcef-17;
+
+  kotlin' = kotlin.overrideAttrs rec {
+    version = versions.${buildType}.kotlinc.version;
     src = fetchurl {
-      url = oldAttrs.src.url;
-      sha256 = "sha256-A1LApFvSL4D2sm5IXNBNqAR7ql3lSGUoH7n4mkp7zyo=";
+      url = "https://cache-redirector.jetbrains.com/maven.pkg.jetbrains.space/kotlin/p/kotlin/kotlin-ide-plugin-dependencies/org/jetbrains/kotlin/kotlin-dist-for-ide/${version}/kotlin-dist-for-ide-${version}.jar";
+      sha256 = versions.${buildType}.kotlinc.hash;
     };
-  });
+    unpackPhase = ''
+      runHook preUnpack
 
-  jbr = jetbrains.jdk-no-jcef-17;
+      LANG=en_US.UTF-8 unzip -qq "$src"
+
+      runHook postUnpack
+    '';
+  };
 
   ideaSrc = fetchFromGitHub {
     owner = "jetbrains";
     repo = "intellij-community";
     rev = "${buildType}/${buildNumber}";
-    hash = ideaHash;
+    sha256 = ideaHash;
```

Comment:
The preferred key is `hash`. Please do not change these back to the deprecated `sha256` key.

Reply:
Already refactored.